src/components/AnnotationShareControl.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  CopyIcon,
+  IconButton,
+  Input,
+  InputGroup,
+  Popover,
+  ShareIcon,
+} from '@hypothesis/frontend-shared';
+import { useCallback, useEffect, useRef, useState } from 'preact/hooks';
+
+import type { Group } from '../helpers';
+import type { Annotation } from '../helpers/annotation-metadata';
+import { isPrivate } from '../helpers/permissions';
+import { isIOS } from '../utils/user-agent';
+
+export type AnnotationShareControlProps = {
+  /** The annotation in question */
+  annotation: Annotation;
+  /** Group to which the annotation belongs */
+  group: Group | null;
+
+  /**
+   * Invoked when the URI is copied to the clipboard.
+   * It indicates if copying to the clipboard was successful or not.
+   */
+  onCopy?: (result: { successful: boolean }) => void;
```

Comment:
We should probably standardize on the `Result<T, E>` type used in `src/types/port-rpc.calls.d.ts` in the client to represent results as sync values (in async contexts, we'd use a promise).

Reply:
Yep, good idea.